src/legacy/core_plugins/data/public/index.ts

```diff
@@ -38,12 +38,16 @@ class DataService {
 }
 
 /**
- * We temporarily export default here so that users importing from 'plugins/data'
+ * We export data here so that users importing from 'plugins/data'
  * will automatically receive the response value of the `setup` contract, mimicking
  * the data that will eventually be injected by the new platform.
  */
-// eslint-disable-next-line import/no-default-export
-export default new DataService().setup();
+export const data = new DataPlugin().setup();
 
 /** @public */
-export type DataSetup = ReturnType<DataService['setup']>;
+export interface DataSetup {
+  indexPatterns: IndexPatternsSetup;
+}
+
+/** @public types */
+export { IndexPattern, StaticIndexPattern, StaticIndexPatternField, Field } from './index_patterns';
```

Comment:
I did it here because the guidance from the Core team was to export all types at the top level (so downstream user's don't need to know about the paths to the subcomponents of your plugin). I suppose we could use a similar pattern at the service-level if we wanted to, the only reason I didn't was because of `.d.ts` files: it felt cleaner to only re-export types from the service.

In a service class file, if I were to `export { Foo } from 'ui/foo';` and `Foo` is both a type and a module, I'll be exporting both from my service. But if I explicitly `import * as types from 'ui/foo';` and `export type Foo = types.Foo;` it ensures that only the type is being exported.

Related: There was [discussion of potentially namespacing types by service in Core](https://github.com/elastic/kibana/issues/33963) (e.g. `IndexPatterns.Field`), but they haven't done this yet because the current thinking is that, over time, _most_ types people will need to import will already be included in a plugin's Setup interface (`DataSetup`).

So my current proposed conventions for handling types during migration would be:

- Export the `setup` interface from each service (& `index.ts`), either using `ReturnType` or by explicitly writing the interface
- Import everything into the top-level Plugin service
- Explicitly export the top-level `setup` interface (`DataSetup`)
- Re-export any additional types for services from the top-level

This makes it so that the only place people need to import from is the top-level directory `plugins/data`